### PR TITLE
Fix for PMI assert:  'vnWx != NoVN'

### PR DESCRIPTION
**Pull request by briansull**
This assert can fire when rhsVNPair is still set to NoVN, because we set isNewUniq to true and didn't set up rhsVNPair .
 
This fix reorders the check for isNewUniq, so that we avoid calling vnPairApplySelectorsAssign when we will assign the rhsVNPair to  a new unique value number.

Fixes VSO 750014

### CODE REVIEW

**Comment by briansull** [comment]
@ANDREENKOS PTAL

**Comment by briansull** [comment]
@sandreenko PTAL

**Comment by tannergooding** [comment]
Can we validate this also fixes https://github.com/dotnet/coreclr/issues/21490?

**Comment by briansull** [comment]
Can I run this test leg on my changes using the CI?

`Project x64_checked_windows_nt_jitx86hwintrinsicnosimd`

**Comment by tannergooding** [comment]
test Windows_NT x64 Checked jitx86hwintrinsicnosimd

**Comment by briansull** [comment]
@dotnet-bot Test  Windows_NT x64 Checked jitx86hwintrinsicnosimd

File: src/jit/valuenum.cpp [hunk 1]
```diff
@@ -6883,7 +6887,6 @@ void Compiler::fgValueNumberTree(GenTree* tree)
         switch (oper)
         {
             case GT_LCL_VAR:
-            case GT_REG_VAR:
```
Comment (sandreenko) [inline src/jit/valuenum.cpp]: Why do you delete this?

**Comment by sandreenko** [comment]
cc @dotnet/jit-contrib

File: src/jit/valuenum.cpp (hunk 1 above)
Comment (briansull) [reply src/jit/valuenum.cpp]: I rebased and pushed to origin to fix this

**Comment by briansull** [comment]
@dotnet-bot Test Windows_NT x64 Checked jitx86hwintrinsicnosimd

**Comment by briansull** [comment]
@tannergooding  - Yes, this change also is fixing  #21490

See
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_windows_nt_jitx86hwintrinsicnosimd_prtest/

**Comment by briansull** [comment]
Also fixes #21231

**Pull request merged**